app/src/main/java/de/tum/in/tumcampusapp/utils/NetUtils.kt

```diff
@@ -0,0 +1,53 @@
+package de.tum.`in`.tumcampusapp.utils
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.NetworkCapabilities
+import android.net.NetworkCapabilities.NET_CAPABILITY_INTERNET
+import android.net.NetworkCapabilities.NET_CAPABILITY_VALIDATED
+import android.os.Build
+
+object NetUtils {
+
+    private fun NetworkCapabilities.hasConnectivity(): Boolean {
+        val capability = if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) NET_CAPABILITY_INTERNET else NET_CAPABILITY_VALIDATED
+        return hasCapability(capability)
+    }
+
+    private inline fun anyConnectedNetwork(context: Context, condition: (NetworkCapabilities) -> Boolean): Boolean {
+        val connectivityMgr = context.getSystemService(Context.CONNECTIVITY_SERVICE) as? ConnectivityManager
+                ?: return false
+
+        return connectivityMgr.allNetworks.asSequence().filter {
+            connectivityMgr.getNetworkInfo(it)?.isConnected ?: false
+        }.mapNotNull {
+            connectivityMgr.getNetworkCapabilities(it)
+        }.any {
+            condition(it)
+        }
+    }
+
+    /**
+     * Check if a network connection is available or can be available soon
+     *
+     * @return if any internet connectivity is available
+     */
+    @JvmStatic
+    fun isConnected(context: Context): Boolean {
+        return anyConnectedNetwork(context) { it.hasConnectivity() }
```

Comment:
We pass `NetworkCapabilities.hasConnectivity()` in both cases. Can you move this to `anyConnectedNetwork()` directly?

Reply:
Then calling `anyConnectedNetworkWithConnectivity()`?
Honestly I think sometimes one has to stop with DRY :see_no_evil: 
I'd rather not have an overly specialized function for this

Reply:
Semantically, it’s confusing to me. Calling `anyConnectedNetwork()` should return whether there is any connected network without requiring a `NetworkCapabilities.hasConnectivity()` condition. After all, the name of the method implies that it already does that.